ParkingZoneApp.Tests/Controllers/Admin/ParkingZoneControllerUnitTests.cs

```diff
@@ -0,0 +1,317 @@
+            _parkingZoneServiceMock.Verify(x => x.GetById(parkingZone.Id), Times.Once);
+        }
+        #endregion
+
+        #region Create
+        [Fact]
+        public void GivenValidCreateVM_WhenPostCreateIsCalled_ThenModelStateIsTrueAndReturnsRedirectToIndex()
+        {
+            //Arrange
+            var createVM = new CreateVM()
+            {
+                Id = Guid.NewGuid(),
+                Name = "Test Name",
+                Address = "Test Address",
+                CreatedDate = new DateOnly(2024, 7, 7)
+            };
+
+            _parkingZoneServiceMock 
+                    .Setup(x => x.Insert(parkingZone));
+
+            //Act
+            var result = _controller.Create(createVM);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.Equal("Index", ((RedirectToActionResult)result).ActionName);
+            Assert.True(_controller.ModelState.IsValid);
+            _parkingZoneServiceMock.Verify(service => service.Insert(It.IsAny<ParkingZone>()), Times.Once);
+        }
+
+        [Theory]
+        [InlineData("Name", "Required")]
+        [InlineData("Address", "Required")]
+        public void GivenCreateVM_WhenPostCreateIsCalled_ThenModelStateIsFalseAndReturnsViewResult(string key, string errorMessage)
+        {
+            //Arrange
+            CreateVM createVM = new CreateVM()
+            {
+                Id = parkingZone.Id,
+                Name = parkingZone.Name,
+                Address = parkingZone.Address,
+                CreatedDate = parkingZone.CreatedDate
+            };
+
+            _controller.ModelState.AddModelError(key, errorMessage);
+    
+            //Act
+            var result = _controller.Create(createVM);
+
+            //Assert
+            var model = Assert.IsType<ViewResult>(result).Model;
+            Assert.IsAssignableFrom<CreateVM>(model);
+            Assert.False(_controller.ModelState.IsValid);
+            Assert.Equal(JsonSerializer.Serialize(createVM), JsonSerializer.Serialize(model));
+        }
+
+        [Fact]
+        public void GivenNothing_WhenGetCreateIsCalled_ReturnsViewResult()
+        {
+            //Arrange
+            _parkingZoneServiceMock
+                    .Setup(x => x.GetById(parkingZone.Id))
+                    .Returns(parkingZone);
+
+            //Act
+            var result = _controller.Create();
+
+            //Assert
+            Assert.IsAssignableFrom<ViewResult>(result);
+            Assert.IsType<ViewResult>(result);
+        }
+        #endregion
+
+        #region Edit
+        [Fact]
+        public void GivenValidIDAndEditVM_WhenPostEditIsCalled_ThenModelStateIsTrueReturnsRedirectToIndex()
+        {
+            //Arrange
+            EditVM editVM = new EditVM()
+            {
+                Id = parkingZone.Id,
+                Name = parkingZone.Name,
+                Address = "New Address",
+                CreatedDate = parkingZone.CreatedDate,
+            };
+
```

Comment:
how your test is working without _parkingZoneService.Update(parkingZone);

Reply:
previously I had that too but I removed it then test was still passing so I thought I dont need it